crates/sui-graphql-e2e-tests/tests/consistency/dynamic_fields.move

```diff
@@ -0,0 +1,549 @@
+    }
+
+    public fun mutate_child(child: &mut Child) {
+        child.count = child.count + 1;
+    }
+
+    public fun mutate_child_via_parent(parent: &mut Parent, name: u64) {
+        mutate_child(ofield::borrow_mut(&mut parent.id, name))
+    }
+
+    public fun reclaim_child(parent: &mut Parent, name: u64): Child {
+        ofield::remove(&mut parent.id, name)
+    }
+
+    public fun delete_child(parent: &mut Parent, name: u64) {
+        let Child { id, count: _ } = reclaim_child(parent, name);
+        object::delete(id);
+    }
+
+    public entry fun add_df(obj: &mut Parent) {
+        let id = &mut obj.id;
+        field::add<String, String>(id, utf8(b"df1"), utf8(b"df1"));
+        field::add<String, String>(id, utf8(b"df2"), utf8(b"df2"));
+        field::add<String, String>(id, utf8(b"df3"), utf8(b"df3"));
+    }
+
+    public entry fun remove_df(obj: &mut Parent) {
+        let id = &mut obj.id;
+        field::remove<String, String>(id, utf8(b"df1"));
+        field::remove<String, String>(id, utf8(b"df2"));
+        field::remove<String, String>(id, utf8(b"df3"));
+    }
+
+    public entry fun add_more_df(obj: &mut Parent) {
+        let id = &mut obj.id;
+        field::add<String, String>(id, utf8(b"df4"), utf8(b"df4"));
+        field::add<String, String>(id, utf8(b"df5"), utf8(b"df5"));
+        field::add<String, String>(id, utf8(b"df6"), utf8(b"df6"));
+    }
+}
+
+//# programmable --sender A --inputs @A 42
+//> 0: Test::M1::parent(Input(0));
+//> 1: Test::M1::child(Input(0));
+
+//# view-object 2,1
+
+//# view-object 2,0
+
+//# programmable --sender A --inputs object(2,1) object(2,0) 420
+//> Test::M1::add_child(Input(0), Input(1), Input(2));
+//> Test::M1::mutate_child_via_parent(Input(0), Input(2));
+
+//# view-object 2,1
+
+//# view-object 2,0
+
+//# create-checkpoint
+
+//# run-graphql
+fragment DynamicFieldSelect on DynamicField {
+  name {
+    bcs
+    type {
+      repr
+    }
+  }
+  value {
+    ... on MoveObject {
+      contents {
+        json
+      }
+    }
+    ... on MoveValue {
+      json
+    }
+  }
+}
+
+fragment DynamicFieldsSelect on DynamicFieldConnection {
+  edges {
+    cursor
+    node {
+      ...DynamicFieldSelect
+    }
+  }
+}
+
+fragment ParentSelect on Parent {
+  parent {
+    dynamicFields {
+      ...DynamicFieldsSelect
+    }
+  }
+}
+
+{
+  parent_version_2_no_dof_but_will_show_dof: object(address: "@{obj_2_1}", version: 2) {
+    address
+    dynamicFields {
+      ...DynamicFieldsSelect
+    }
+  }
+  parent_version_3_has_dof: object(address: "@{obj_2_1}", version: 3) {
+    dynamicFields {
+      ...DynamicFieldsSelect
+    }
+    child_dof_state_consistent: dynamicObjectField(name: {type: "u64", bcs: "pAEAAAAAAAA="}) {
+      value {
+        ... on MoveObject {
+          address
+          # Note that the following will be incorrect until fixed - we need to look up the object's
+          # contents using its df_object_id, but its parent is technically the wrapper's parent
```

Comment:
I'm not following -- can you explain in what way the results are incorrect (in terms of what we see in the results), and add some more detail on what the fix should be. Is this something we are fixing straight away, or something that is longer term?

Reply:
when we look up a dynamic object field, the object retrieved is the Field<Wrapper<K>, V> which points to the expected parent object. To resolve the value of this dynamic object field, we look up the wrapped objec for the actual contents of the dynamic object field. This move object will have its parent pointed to the wrapper, not the actual parent object. An easy change would be to `MoveObject::query` the `df_object_id`, and then replace the stored object's `owner` field with the expected owner

This does not apply to dynamic fields since we resolve its value as a `MoveValue`